test/toJSON.test.js

```diff
@@ -172,3 +172,1020 @@ test('throw an error on non nullable null object', (t) => {
+  const arrayStringsAndNumberNumber = {
+    arrayOfStringsAndNumbers: { toJSON () { return [dataNumber1, dataNumber2] } }
+  }
+  const arrayStringsAndNumberMixed = {
+    arrayOfStringsAndNumbers: { toJSON () { return [dataString1, dataNumber1, dataNumber2, dataString2] } }
+  }
+  const arrayStringsAndNumberMixedNull = {
+    arrayOfStringsAndNumbers: { toJSON () { return [dataString1, dataNull, dataNumber1, dataNumber2, dataString2, dataNull] } }
+  }
+
+  try {
+    const value = stringify({ arrayOfStringsAndNumbers: dataNull })
+    t.equal(value, '{"arrayOfStringsAndNumbers":null}')
+  } catch (e) {
+    console.log(e)
+    t.fail()
+  }
+
+  try {
+    const value = stringify(arrayStringsAndNumberString)
+    t.equal(value, '{"arrayOfStringsAndNumbers":["string1","string2"]}')
+  } catch (e) {
+    console.log(e)
+    t.fail()
+  }
+
+  t.equal(stringify(arrayStringsAndNumberNumber), '{"arrayOfStringsAndNumbers":[42,7]}')
+  t.equal(stringify(arrayStringsAndNumberMixed), '{"arrayOfStringsAndNumbers":["string1",42,7,"string2"]}')
+  t.equal(stringify(arrayStringsAndNumberMixedNull), '{"arrayOfStringsAndNumbers":["string1",null,42,7,"string2",null]}')
+})
+
+test('object that is simultaneously a string and a json', (t) => {
+  t.plan(2)
+  const schema = {
+    type: 'object',
+    properties: {
+      simultaneously: {
+        type: ['string', 'object'],
+        properties: {
+          foo: { type: 'string' }
+        }
+      }
+    }
+  }
+
+  const likeObjectId = {
+    toString () { return 'hello' }
+  }
+
+  const stringify = build(schema)
+  const valueStr = stringify({ simultaneously: { toJSON () { return likeObjectId } } })
+  t.equal(valueStr, '{"simultaneously":"hello"}')
+
+  const valueObj = stringify({ simultaneously: { toJSON () { return { foo: likeObjectId } } } })
+  t.equal(valueObj, '{"simultaneously":{"foo":"hello"}}')
+})
+
+test('object that is simultaneously a string and a json switched', (t) => {
+  t.plan(2)
+  const schema = {
+    type: 'object',
+    properties: {
+      simultaneously: {
+        type: ['object', 'string'],
+        properties: {
+          foo: { type: 'string' }
+        }
+      }
+    }
+  }
+
+  const likeObjectId = {
+    toString () { return 'hello' }
+  }
+
+  const stringify = build(schema)
+  const valueStr = stringify({ simultaneously: { toJSON () { return likeObjectId } } })
+  t.equal(valueStr, '{"simultaneously":{}}')
+
+  const valueObj = stringify({ simultaneously: { toJSON () { return { foo: likeObjectId } } } })
+  t.equal(valueObj, '{"simultaneously":{"foo":"hello"}}')
+})
+
+test('should throw an error when type is array and object is null', (t) => {
+  t.plan(1)
+  const schema = {
+    type: 'object',
+    properties: {
+      arr: {
+        type: 'array',
+        items: {
+          type: 'number'
+        }
+      }
+    }
+  }
+
+  const dataNull = { toJSON () { return null } }
+
+  const stringify = build(schema)
+  t.throws(() => stringify({ arr: dataNull }))
```

Comment:
Which error?

Reply:
'The value \'null\' does not match schema definition.'